client.go

```diff
@@ -800,3 +822,62 @@ func (c *TunaSessionClient) removeClosedSessions() {
 		c.Unlock()
 	}
 }
+
+func (c *TunaSessionClient) ListenUDP(network string, laddr *net.UDPAddr) (*udpconn.EncryptUDPConn, error) {
+	conn, err := net.ListenUDP(network, laddr)
+	if err != nil {
+		return nil, err
+	}
+	return udpconn.NewEncryptUDPConn(conn), nil
+}
+
+func (c *TunaSessionClient) DialUDP(remoteAddr string) (*udpconn.EncryptUDPConn, error) {
+	config, err := nkn.MergeDialConfig(c.config.SessionConfig, nil)
+	if err != nil {
+		return nil, err
+	}
+	return c.DialUDPWithConfig(remoteAddr, config)
+}
+
+func (c *TunaSessionClient) DialUDPWithConfig(remoteAddr string, config *nkn.DialConfig) (*udpconn.EncryptUDPConn, error) {
+	config, err := nkn.MergeDialConfig(c.config.SessionConfig, config)
+	if err != nil {
+		return nil, err
+	}
+
+	pubAddrs, err := c.getPubAddrsFromRemote(remoteAddr, config)
+	if err != nil {
+		return nil, err
+	}
+
+	udpConn := new(udpconn.EncryptUDPConn)
+	addr := pubAddrs.Addrs[len(pubAddrs.Addrs)-1]
+	udpAddr := net.UDPAddr{IP: net.ParseIP(addr.IP), Port: int(addr.Port)}
+	conn, err := net.DialUDP("udp", nil, &udpAddr)
+	if err != nil {
+		log.Printf("dial UDP error: %v", err)
+		return nil, err
+	}
+	udpConn = udpconn.NewEncryptUDPConn(conn)
+	host, portStr, _ := net.SplitHostPort(udpConn.LocalAddr().String())
+	port, _ := strconv.Atoi(portStr)
+	err = udpConn.AddCodec(&net.UDPAddr{IP: net.ParseIP(host), Port: port}, c.sharedKeys[remoteAddr], tpb.EncryptionAlgo_ENCRYPTION_XSALSA20_POLY1305, false)
+	if err != nil {
+		log.Printf("add codec err: %v", err)
+		return nil, err
+	}
+
+	return udpConn, nil
+}
+
+func (c *TunaSessionClient) ServicePort() int {
+	return int(c.tunaExits[len(c.tunaExits)-1].ReverseMetadata.ServiceUdp[0])
```

Comment:
Why not use 0? And it seems we don't need this API, is it for debug?

Reply:
> Why not use 0? And it seems we don't need this API, is it for debug?

For high-level usage such as `nkn-tunnel`, need to know the port number to connect